src/app/DownloadsDialog.qml

```diff
@@ -0,0 +1,190 @@
+ */
+
+import QtQuick 2.9
+import QtWebEngine 1.9
+import Ubuntu.Components 1.3
+import Ubuntu.Components.Popups 1.3
+import QtQuick.Layouts 1.3
+import Ubuntu.Content 1.3
+import webbrowsercommon.private 0.1
+
+import "MimeTypeMapper.js" as MimeTypeMapper
+import "FileUtils.js" as FileUtils
+
+Popover {
+    id: downloadsDialog
+
+    property var downloadsList: []
+    property bool isEmpty: downloadsListView.count === 0
+    readonly property real controlsHeight: (downloadsDialogColumn.spacing * (downloadsDialogColumn.children.length - 1)) 
+                                            + (downloadsDialogColumn.anchors.margins * 2) + buttonsBar.height + titleLabel.height
+    property real maximumWidth: units.gu(60)
+    property real preferredWidth: browser.width - units.gu(6)
+    
+    property real maximumHeight: browser.height - units.gu(6)
+    property real preferredHeight:  downloadsDialogColumn.height + units.gu(2)
+    
+    signal showDownloadsPage()
+    signal preview(string url)
+
+    contentHeight: preferredHeight > maximumHeight ? maximumHeight : preferredHeight
+    contentWidth: preferredWidth > maximumWidth ? maximumWidth : preferredWidth
+    
+    grabDismissAreaEvents: true
+
+    ColumnLayout {
+        id: downloadsDialogColumn
+        
+        anchors {
+            top: parent.top
+            left: parent.left
+            right: parent.right
+            margins: units.gu(1)
+        }
+
+        spacing: units.gu(2)
+
+        Label {
+            id: titleLabel
+
+            Layout.fillWidth: true
+            font.bold: true
+            text: i18n.tr("Recent Downloads")
+        }
+        
+        Rectangle {
+            id: emptyLabel
+            
+            visible: isEmpty
+            color: "transparent"
+
+            Layout.preferredHeight: units.gu(10)
+            Layout.minimumHeight: label.height
+            Layout.fillWidth: true
+            
+            Label {
+                id: label
+                text: i18n.tr("No Recent Downloads")
+                anchors.centerIn: parent
+            }
+        }
+
+        ListView {
+            id: downloadsListView
+            
+            visible: !isEmpty
+            Layout.fillWidth: true
+            Layout.preferredHeight: downloadsListView.count * units.gu(7)
+            Layout.maximumHeight: browser.height - controlsHeight - units.gu(7)
+            Layout.minimumHeight: units.gu(7)
+            clip: true
+
+            model: downloadsList
+
+            property int selectedIndex: -1
+            
+            delegate: SimpleDownloadDelegate {
+                download: modelData
+                title.text: FileUtils.getFilename(modelData.path)
+                subtitle.text: if (cancelled) {
+                            i18n.tr("Cancelled")
+                        } else {
+                            (error ? modelData.interruptReasonString : (incomplete ? i18n.tr("%1%").arg(progress) : i18n.tr("Completed")))
```

Comment:
This appears to be another instance where no translatable text is provided inside `i18n.tr()`.

Reply:
I can't remember completely anymore why I did this but I think this is because different languages writes percentage differently like for example Arabic it's reversed [https://en.wikipedia.org/wiki/Percent_sign]